akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala

```diff
@@ -392,7 +392,7 @@ object HttpEntity {
       if (contentType == this.contentType) this else copy(contentType = contentType)
 
     override def withSizeLimit(maxBytes: Long): HttpEntity.Default =
-      copy(data = data withAttributes Attributes(SizeLimit(maxBytes, Some(contentLength))))
+      copy(data = limitableByteSource(data) withAttributes Attributes(SizeLimit(maxBytes, Some(contentLength))))
```

Comment:
For `Strict.withSizeLimit` in line 337 you can remove the extra `limitableByteSource` because `withSizeLimit` is called afterwards (whether delegating to Default is the right approach for Strict entities is another question...).